squidDrone/src/i2c/i2c_interface.hpp

```diff
@@ -0,0 +1,47 @@
+#ifndef SRC_I2C_INTERFACE_HPP_
+#define SRC_I2C_INTERFACE_HPP_
+
+#include "i2c_status.hpp"
+#include <stdint.h>
+#include <vector>
+#include <tuple>
+
+namespace i2c {
+
+  /**
+   * @brief Interface of I2C package
+   * 
+   */
+  class I2CInterface {
+    public:
+      virtual ~I2CInterface() = default;
+      explicit I2CInterface(void){};
+
+      /**
+       * @brief Reads data from I2C Bus
+       * 
+       * @param address The address of the I2C Bus participant
+       * @param byte_size Amount of Bytes to read
+       * @param timeout Timeout in milliseconds
+       * @return std::tuple<I2CStatus, std::vector<uint8_t>> Status of I2C Interface and Data read as std::vector. Each element is a Byte read.
+       */
+      virtual auto Read(uint8_t address, uint16_t byte_size, uint32_t timeout=I2C_STANDARD_TIMEOUT_IN_MS) noexcept -> std::tuple<I2CStatus, std::vector<uint8_t>> = 0;
+      
+      /**
+       * @brief Writes data to I2C participant
+       * 
+       * @param address The address of the I2C Bus participant
+       * @param data Data to write to I2C participant. std::vector with each element is one Byte.
+       * @param timeout Timeout in milliseconds
+       * @return I2CStatus Status of I2C Interface
```

Comment:
I added EXTRACT_ALL in Doxyfile.in (https://github.com/elheck/Squiddrone/pull/219/commits/4a12ee03bc79d13370c705dd5fd7bd619ac73947) so that every enum class gets also documented. This is also helpful for HalError etc. 

So now a link to I2CStatus is added:
![Bildschirmfoto vom 2020-05-31 12-43-45](https://user-images.githubusercontent.com/48162347/83350437-6c62c900-a33c-11ea-974c-dd6d7196a114.png)
 
Also I added some documentation in I2CStatus itself:
![Bildschirmfoto vom 2020-05-31 12-44-51](https://user-images.githubusercontent.com/48162347/83350459-92886900-a33c-11ea-9821-2819aa16b14c.png)